drivers/ethernet/dsa_ksz8794.c

```diff
@@ -0,0 +1,579 @@
+	dsa_ksz8794_write_reg(sdev, KSZ8794_REG_IND_DATA_5, p[5]);
+	dsa_ksz8794_write_reg(sdev, KSZ8794_REG_IND_DATA_4, p[4]);
+	dsa_ksz8794_write_reg(sdev, KSZ8794_REG_IND_DATA_3, p[3]);
+	dsa_ksz8794_write_reg(sdev, KSZ8794_REG_IND_DATA_2, p[2]);
+	dsa_ksz8794_write_reg(sdev, KSZ8794_REG_IND_DATA_1, p[1]);
+	dsa_ksz8794_write_reg(sdev, KSZ8794_REG_IND_DATA_0, p[0]);
+
+	dsa_ksz8794_write_reg(sdev, KSZ8794_REG_IND_CTRL_0, 0x00);
+	dsa_ksz8794_write_reg(sdev, KSZ8794_REG_IND_CTRL_1, entry_addr);
+
+	return 0;
+}
+
+static int dsa_ksz8794_set_static_mac_table(struct dsa_ksz8794_spi *sdev,
+					    const uint8_t *mac, uint8_t fw_port,
+					    uint16_t entry_idx)
+{
+	uint8_t buf[8];
+
+	buf[7] = 0;
+	/* Prepare entry for static MAC address table */
+	buf[5] = mac[0];
+	buf[4] = mac[1];
+	buf[3] = mac[2];
+	buf[2] = mac[3];
+	buf[1] = mac[4];
+	buf[0] = mac[5];
+
+	buf[6] = fw_port;
+	buf[6] |= KSZ8794_STATIC_MAC_TABLE_VALID;
+	buf[6] |= KSZ8794_STATIC_MAC_TABLE_OVERRIDE;
+
+	dsa_ksz8794_write_static_mac_table(sdev, entry_idx, buf);
+
+	return 0;
+}
+
+static int dsa_ksz8794_read_static_mac_table(struct dsa_ksz8794_spi *sdev,
+					      uint16_t entry_addr, uint8_t *p)
+{
+	/*
+	 * According to KSZ8794 manual - read from static mac address table
+	 * requires reads from indirect registers:
+	 *
+	 * Write to Register 110 with 0x10 (read static table selected)
+	 * Write to Register 111 with 0x0x (trigger the read operation, to
+	 * table entry x)
+	 *
+	 * Then:
+	 * Write register 0x71 (113)
+	 * ....
+	 * Write register 0x78 (120)
+	 *
+	 */
+
+	dsa_ksz8794_write_reg(sdev, KSZ8794_REG_IND_CTRL_0, 0x10);
+	dsa_ksz8794_write_reg(sdev, KSZ8794_REG_IND_CTRL_1, entry_addr);
+
+	dsa_ksz8794_read_reg(sdev, KSZ8794_REG_IND_DATA_7, &p[7]);
+	dsa_ksz8794_read_reg(sdev, KSZ8794_REG_IND_DATA_6, &p[6]);
+	dsa_ksz8794_read_reg(sdev, KSZ8794_REG_IND_DATA_5, &p[5]);
+	dsa_ksz8794_read_reg(sdev, KSZ8794_REG_IND_DATA_4, &p[4]);
+	dsa_ksz8794_read_reg(sdev, KSZ8794_REG_IND_DATA_3, &p[3]);
+	dsa_ksz8794_read_reg(sdev, KSZ8794_REG_IND_DATA_2, &p[2]);
+	dsa_ksz8794_read_reg(sdev, KSZ8794_REG_IND_DATA_1, &p[1]);
+	dsa_ksz8794_read_reg(sdev, KSZ8794_REG_IND_DATA_0, &p[0]);
+
+	return 0;
+}
+
+static int dsa_ksz8794_get_static_mac_table(struct dsa_ksz8794_spi *sdev,
+					    uint8_t *buf, uint16_t entry_idx)
+{
+	return dsa_ksz8794_read_static_mac_table(sdev, entry_idx, buf);
+}
+
+static int dsa_ksz8794_switch_setup(struct dsa_ksz8794_spi *sdev)
+{
+	uint8_t tmp, i;
+
+	/*
+	 * Loop through ports - The same setup when tail tagging is enabled or
+	 * disabled.
+	 */
+	for (i = KSZ8794_PORT1; i <= KSZ8794_PORT3; i++) {
+		/* Enable transmission, reception and switch address learning */
+		dsa_ksz8794_read_reg(sdev, KSZ8794_CTRL2_PORTn(i), &tmp);
+		tmp |= KSZ8794_CTRL2_TRANSMIT_EN;
+		tmp |= KSZ8794_CTRL2_RECEIVE_EN;
+		tmp &= ~KSZ8794_CTRL2_LEARNING_DIS;
+		dsa_ksz8794_write_reg(sdev, KSZ8794_CTRL2_PORTn(i), tmp);
+	}
+
+#if IS_ENABLED(CONFIG_DSA_KSZ8794_TAIL_TAGGING)
```

Comment:
We use IS_ENABLED() for C if, not pre-processor #if. 
(unless it's accepted to use IS_ENABLED, but the purpose of that macros was to avoid pre-processor #if )

Used defined() here. Or switch to an actual C if/else with IS_ENABLED

Reply:
Ok. Thanks for pointing this out.